### ISSUE TITLE
FxA support

### ISSUE THREAD

**Issue by ianb** [mozilla-services/screenshots#3080]
These are requirements for FxA:

I propose we land this as two chunks, the first:

- [ ] Can access a /settings page from My Shots
- [ ] When you are not logged into FxA, a button is presented to connect / add an account
- [ ] Clicking that button puts you through the OAuth FxA flow
- [ ] If you cancel the login flow (on the FxA site), you should return to /settings
- [ ] If you complete the flow, you should end up back at /settings
- [ ] If you are in /settings and are logged in, you should see your avatar image, and your name or email
- [ ] We should store the avatar image, name, and email in the database (note: we initially left out the email, but we should include it as well)
- [ ] There should be a `x` button on the settings page, that redirects the user back to `/shots`
- [ ] If you are not logged in and try to access `/settings` you should get permission denied
- [ ] If you are in /settings and logged in, then there should be a Disconnect button
- [ ] The Disconnect button should erase `devices.accountid` for the current login (`req.deviceId`)
- [ ] There should be CSRF protection on Disconnect (see `/api/set-expiration` for an example); there does not have to be protection on login/connect
- [ ] The settings page and the link to the settings page should be controlled by a configuration variable (defined in `config.js`). We should default it to off. We can ask about turning it on just for dev.

Then as a follow-on:

- [ ] When you are logged in, you should be able to access shots made on associated devices as though you own them
- [ ] You should be able to delete shots, set their expiration, and set their title

**Comment by youwenliang** [comment]
@niharikak101 
Here's the UX measurement:
https://mozilla.github.io/testpilot-assets/Firefox_Screenshots/FxA_Integration/

**Comment by youwenliang** [comment]
@ianb 
How are we implementing the disconnect behavior now? Are we keeping the shots that are not connected to an account, or are we removing all the shots once a user disconnected?

**Comment by niharikak101** [comment]
@youwenliang We are removing the shots associated with the account that don't belong to the device which the user disconnects from the account, but keeping the shots that were taken from that specefic device/profile. Essentially, when a user disconnects they can't access the shots that were taken from other devices associated with the account.

I've also added some screenshots of what the settings page looks like right now. @ianb and I talked about saying 'connect' instead of sign in, and I've removed the Manage Accounts button because we don't have any functionality for that, yet.

Connected Account:
![screenshot-2017-7-13 settings](https://user-images.githubusercontent.com/16074636/28150134-499cb162-67b0-11e7-9625-d581ced1749d.png)

Guest Account:
<img width="1192" alt="screen shot 2017-07-13 at 09 34 09" src="https://user-images.githubusercontent.com/16074636/28150001-51213792-67af-11e7-946a-5893122bf21f.png">
